Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/tasks/service/PersistedTasksServiceTest.kt

```diff
@@ -81,15 +86,18 @@ class PersistedTasksServiceTest {
 
     @Test
     fun given_TheLocalDataHasTasksAndTasksAreFresh_on_GetTasksEvents_it_ShouldReturnTasksFromTheLocalData() {
+        val testScheduler = givenPresenterWithTestScheduler()
         val remoteTasks = sampleRemoteTasks()
         val localTasks = sampleLocalTasks()
         putTaskListIntoRemoteDataSource(remoteTasks)
         putTasksIntoLocalDataSource(localTasks)
         `when`(freshnessChecker.isFresh(localTasks)).thenReturn(true)
 
         val testObserver = subscribeToTasksEvent()
+        testScheduler.triggerActions()
 
         testObserver.assertReceivedOnNext(listOf(
+                emptyIdleEvent(),
```

Comment:
With the testScheduler it will get this item since before it evens performs any action the original value is sent. Before since it was running the observable instantly it was not sent.

This is not a change in behaviour, just in the tests, we now get the full behaviour here to test => better tests.